### ISSUE TITLE
Make Item Holders even better

### ISSUE THREAD

**Issue by lailani-f** [GTNewHorizons/GT-New-Horizons-Modpack#17770]
### Your GTNH Discord Username

cure.starlight

### Your Pack Version

2.7.0-beta3

### Your Proposal

The new Item Holders are great and let you store lenses/extruder shapes etc. right inside the machine. However accessing stored items is a bit cumbersome as the holder needs to be opened by clicking on its cover icon inside the machine GUI every time the machine is accessed. Furthermore the holder GUI opens far off to the side of the machine GUI which does not seem ideal. It can be manually moved around but will not remember its position once the machine GUI is closed and reopened. Lastly once the holder GUI is open it will intercept the ESC key used to usually close the whole machine GUI, making it necessary to hit ESC twice to reach the desired result.

I therefore propose the Item Handler GUI to remember its position and opened state and to ignore ESC keypress.

[Item Holder GUI opens far to the side.]
![image](https://github.com/user-attachments/assets/81244f6a-325b-4043-b126-e265d6f351d9)

### Your Goal

Make the Item Holder even better by making its use more intuitive and seemless.

### Your Vision

A smoll QoL change surely noone can resist!

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying *I like this change please reconsider adding it* will prompt us to investigate and reopen it once we confirm your report.
- [X] I understand this change request may not attract enough attention and thus not be implemented.
- [X] I understand this change request may be rejected due to other community members think it's inappropriate.
- [X] I believe this feature would make the pack better.

**Comment by Rika-TH** [comment]
This would be amazing, after we finally found the name of it (might be a bit better as hear there's a quest now) us and our teammate both had this exact type of annoyance, the positioning is so dum

**Comment by Impos913** [comment]
I agree with this suggestion, using the holder is pointless imo if I have to reopen its gui every time i open the machine

**Comment by Glease** [comment]
I agree that this is indeed a problem. here are some thoughts on the implementation
1. make item holder autoopen on gui open.
2. tile the holders to the left of the screen, as close to the main gui as possible
3. remember the location of holder gui on client side

all of these have the same issue: what if there are multiple holder cover here?

**Comment by DownerCase** [comment]
> I agree that this is indeed a problem. here are some thoughts on the implementation
> 
> 1. make item holder autoopen on gui open.
> 2. tile the holders to the left of the screen, as close to the main gui as possible
> 3. remember the location of holder gui on client side
> 
> all of these have the same issue: what if there are multiple holder cover here?

- Default state of the holder should be closed and it will remember its state (position and open/close).
- Default opening position doesn't matter too much.
- Opening a holder will have it render above any already open holders (to prevent it being totally hidden)
- Add an icon that suggests the window can be moved (I had no idea until seeing someone mention it! (Not that it really matters with it currently not saving that position...))

This way the user will know they can move the holders however they want, and won't be confused by a holder opening behind another.

**Comment by feioO** [comment]
Another improvement could be to make shift clicking from the item holder to put said item in the machine instead of the player's inventory. And back from the machine to the holder instead of the inventory.
